### PR TITLE
snow: Add new parameter 'host'

### LINKED ISSUE #62018: snow_record (and snow resources in general) should allow for host as well as instance
##### SUMMARY
snow_record only allows for "instance" parameter to be passed and not host parameter to pysnow.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
snow_record 

##### ADDITIONAL INFORMATION
Pysnow has both an instance and a host parameter be set passed for target snow instance.  Often when internal systems go through adfs/sso options they will be pointing to an internal nat before it gets to the public site which would not be available on 443.  We've got around this by manually hacking the resource, to have the instance ansible parameter point to the host pysnow variable.  This obviously is not a long term solution.

This reflects the documentation at pysnow:
https://pysnow.readthedocs.io/en/latest/api/client.html

```
- name: Grab a user record
  snow_record:
    username: ansible_test
    password: my_password
    host: dev99999.blah.com
    state: present
    number: 62826bf03710200044e0bfc8bcbe5df1
    table: sys_user
    lookup_field: sys_id
```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Added support for specifying custom service now domain.

Fixes: #62018

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE 
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/62018_snow_host.yml
lib/ansible/module_utils/service_now.py
lib/ansible/modules/notification/snow_record.py
lib/ansible/modules/notification/snow_record_find.py
lib/ansible/plugins/doc_fragments/service_now.py

### CODE REVIEW

File: lib/ansible/plugins/doc_fragments/service_now.py [hunk 1]
```diff
@@ -15,8 +15,18 @@ class ModuleDocFragment(object):
...
       - If the value is not specified in the task, the value of environment variable C(SN_INSTANCE) will be used instead.
       - Environment variable support added in Ansible 2.9.
+      - Mutually exclusive with C(host).
       required: false
       type: str
+    host:
+      description:
+      - The ServiceNow hostname.
+      - This value is FQDN for ServiceNow host.
+      - If the value is not specified in the task, the value of environment variable C(SN_HOST) will be used instead.
```
Comment (Andersson007) [inline lib/ansible/plugins/doc_fragments/service_now.py:25]:
```suggestion
      - If the value is not specified in the task, the value of environment variable C(SN_HOST) will be used instead.
      - Environment variable support added in Ansible 2.9.
```

**Review by Andersson007 — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/doc_fragments/service_now.py (hunk 1 above)
Comment (Andersson007) [reply lib/ansible/plugins/doc_fragments/service_now.py:25]: it is mentioned in the ```instanse``` param description, IMO better to mention this here too

Comment (Akasurde) [reply lib/ansible/plugins/doc_fragments/service_now.py:25]: OK.

**Review by Andersson007 — APPROVED** [review]
i don't use this so can't try it manually. would be greate if someone did
shipit

**Comment by Akasurde** [comment]
@garbled1 @n3pjk @jmcleroy  @theemptyone Can you please test this PR and let me know if it works for you? Thanks in advance.

**Comment by martbhell** [comment]
Hello, I tested this in https://github.com/ansible/ansible/issues/41320#issuecomment-582426160

**Comment by Akasurde** [comment]
Migrated to https://github.com/ServiceNowITOM/servicenow-ansible/pull/5

**Pull request closed without merging**